quic/s2n-quic-core/src/recovery/bandwidth/estimator.rs

```diff
@@ -200,6 +207,16 @@ impl Estimator {
         self.delivered_bytes += bytes_acknowledged as u64;
         self.delivered_time = Some(now);
 
+        if self
+            .app_limited_delivered_bytes
+            .map_or(false, |app_limited_bytes| {
+                self.delivered_bytes > app_limited_bytes
+            })
+        {
+            // Clear app-limited field if bubble is ACKed and gone
```

Comment:
It comes from the related [delivery rate estimation draft RFC](https://datatracker.ietf.org/doc/html/draft-cheng-iccrg-delivery-rate-estimation#section-2.3): 
> An application-limited phase starts when the sending application
   requests to send new data, or the connection's retransmission
   mechanisms decide to retransmit data, and the connection meets all of
   the following conditions ... 
> If these conditions are all met then the sender has run out of data
   to feed the network.  This would effectively create a "bubble" of
   idle time in the data pipeline.

This comment specifically is from the pseudocode: https://datatracker.ietf.org/doc/html/draft-cheng-iccrg-delivery-rate-estimation#section-3.3